driver-core/src/main/com/mongodb/client/model/expressions/DateExpression.java

```diff
@@ -20,5 +20,18 @@
  * Expresses a date value.
  */
 public interface DateExpression extends Expression {
+    IntegerExpression year(StringExpression timezone);
```

Comment:
Especially given the `dateToString()` overload below, what's the rationale for not having similar no-arg overloads for all of these?  I can think of two:

1. reduced API surface
2. a conclusion that the default of "UTC" is not useful in practice

Or perhaps it's an oversight

Reply:
`dateToString()` gives a string representation of the date in UTC, which is also a string representation of an instant, because BSON date is actually an instant contrary to its name. I.e., `dateToString()` is equivalent to Java's `Instant.toString`, which is useful and can (and should) be thought of without thinking about timezones.

`year()` on the other hand, makes sense only within a specific timezone, but that may not be obvious to a user. To make it obvious, this method always requires a timezone.

Reply:
That's all true, but does it mean that UTC is not a credible default?  We should assume that query team had a good reason for the default unless we hear otherwise.

Reply:
Given the default server format, which is incorrect in all situations but when the timezone is UTC, and that the Java SE API does not have a similar "UTC is the default timezone when someone wants to extract `year`" (instead, the Java SE API forces one to convert `Instant` to `ZonedDateTime` via `ZonedDateTime.ofInstant(Instant instant, ZoneId zone)`, and then allows one to extract `ChronoUnit.YEARS`), it is not obvious that the query team had this API thought through.

Reply:
This also makes it clear that we are not storing a date-with-timezone. I think forcing the user to be explicit about this will prevent mistakes.

Also, overloads would be easy to add later if we're making a mistake, but the alternative would be difficult to remove.